manager/state/raft/transport/peer.go

```diff
@@ -247,6 +336,17 @@ func (p *peer) handleAddressChange(ctx context.Context) error {
 	return nil
 }
 
+// Function to check if streaming is needed.
+// Returns true if the message type is MsgSnap/MsgApp
+// and size larger than MaxRaftMsgSize.
+func needsStreaming(m *raftpb.Message) bool {
```

Comment:
I would advocate for always using streaming unless the other end doesn't support it.

Reply:
Can you please elaborate on this ?

I thought a bit more about this. If the message is smaller than GrpcMaxMsgSize, it's less efficient to first try to stream and then fall back to a basic send if unsupported. 

Having said that, I do see the value in always streaming, and eventually deprecating the other endpoint, unless there is a performance overhead to streaming a single message.

Can you please also elaborate on this ?

Reply:
I also like the idea of always using streaming. It gives the streaming version of the RPC more real-world coverage instead of only using it in unusual corner cases. It would let us eventually deprecate the non-streaming version.

I don't think the mixed-version scenario is important to optimize.

Reply:
I agree with the point about giving this more real-world coverage. I'll make this change.